app/services/migration/migrators/schedule.rb

```diff
@@ -0,0 +1,74 @@
+module Migration::Migrators
+  class Schedule < Base
+    class << self
+      def model_count
+        ecf_schedules.count
+      end
+
+      def model
+        :schedule
+      end
+
+      def ecf_schedules
+        Migration::Ecf::Finance::Schedule.includes(:cohort, :milestones)
+      end
+
+      def dependencies
+        %i[cohort course]
+      end
+    end
+
+    def call
+      migrate(self.class.ecf_schedules) do |ecf_schedule|
+        ensure_milestone_dates_are_all_the_same!(ecf_schedule)
+
+        course_group = course_groups_by_identifier(ecf_schedule.type)
+
+        unless course_group
+          ecf_schedule.errors.add(:base, "Course group not found for schedule")
+          raise ActiveRecord::RecordInvalid, ecf_schedule
+        end
+
+        ::Schedule.find_or_initialize_by(
+          cohort_id: cohorts_by_start_year[ecf_schedule.cohort.start_year].id,
+          name: ecf_schedule.name,
```

Comment:
`name` should move down to the `.tap`, we want to find/initialize with `identifier`, `cohort` and `cohort_group`